src/dnx/servicing.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+#include "version.h"
+#include "utils.h"
+#include <string>
+#include <fstream>
+#include <algorithm>
+#include "TraceWriter.h"
+
+namespace
+{
+    const std::string runtime_moniker =
+#if defined(CORECLR_WIN)
+        "coreclr"
+#else
+        "clr"
+#endif
+        "-win-"
+#if defined(AMD64)
+        "x64"
+#elif defined(ARM)
+        "arm"
+#else
+        "x86"
+#endif
+        ;
+
+    std::wstring find_runtime_replacement(std::ifstream& input, TraceWriter& trace_writer)
+    {
+        const std::string runtime_qualifier = std::string{ "dnx|" } + runtime_moniker + "|" + ProductVersionStr + "=";
+
+        for (std::string line; std::getline(input, line); )
+        {
+            if (line.compare(0, runtime_qualifier.length(), runtime_qualifier) == 0)
+            {
+                return dnx::utils::to_wstring(line.substr(runtime_qualifier.length()));
```

Comment:
As per our conversation - there can be multiple entries for the same runtime. We will now take the last entry from the index.txt.